pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/BaseFilterOperator.java

```diff
@@ -56,4 +73,8 @@ public final BaseFilterBlock getNextBlock(BlockId blockId) {
   }
 
   public abstract BaseFilterBlock nextFilterBlock(BlockId blockId);
+
+  public boolean alwaysFalse() {
```

Comment:
its ok to have a base implementation that returns false. But there is no need to do predicateEvaluator.alwaysFalse() in the base class right. This will avoid having a bunch of new setter methods on filteroperator.